### PR TITLE
 JAMES-2586 Implement PostgresMessageIdMapper

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    private Mono<Pair<Flags, ComposedMessageIdWithMetaData>> updateFlags(Flags newState, MessageManager.FlagsUpdateMode updateMode, PostgresMailboxId mailboxId, ComposedMessageIdWithMetaData oldComposedId) {
+        FlagsUpdateCalculator flagsUpdateCalculator = new FlagsUpdateCalculator(newState, updateMode);
+        Flags newFlags = flagsUpdateCalculator.buildNewFlags(oldComposedId.getFlags());
+        if (identicalFlags(oldComposedId, newFlags)) {
+            return Mono.just(Pair.of(oldComposedId.getFlags(), oldComposedId));
+        } else {
+            return modSeqProvider.nextModSeqReactive(mailboxId)
+                .flatMap(newModSeq -> {
+                    // TODO bug?
+                    // ModSeq previousModseq = oldComposedId.getModSeq();
```
Comment (hungphan227) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java:237]: what do you mean here @chibenwa?

Comment (chibenwa) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java:237]: If this works you can remove the comment IMO

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public long size() {
+            return mailboxMessage.getBodyOctets();
+        }
+    };
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java:93]: Could this be part of DelegatingMailboxMessage ?

IE: `mailboxMessage.getBodyByteSource()`

Then we could rely on this in `PostgresMessageIdMapper`, `PostgresMessageMapper`, `CassandraMessageDAOV3` ?

**Review by chibenwa — COMMENTED** [review]
Looks great to me!

Thanks @hungphan227 for the polish :-p

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java (hunk 2 above)
Comment (hungphan227) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java:93]: I see one implementation of MailboxMessage is CassandraMailboxMessage. Do you know that CassandraMessageDAOV3 use DelegatingMailboxMessage or CassandraMailboxMessage?

Comment (chibenwa) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresMessageIdMapper.java:93]: Ok nevermind. Merge I 'll do the refactoring.

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Can you squash the fixup please @hungphan227 ? :)

**Pull request merged**